### PR TITLE
Fixing the tutorial for non-MNIST datasets

### PR DESCRIPTION
**Pull request by aashish-kumar**
The cleverhans_tutorial has been written only for MNIST with extensive hard coding of data-set parameters(like image width/height/channels. I have modified the code to handle other data-sets. The user need to only modify utils_mnist.py for his dataset and rest of the code will work out of box. Also I made sure the tutorials work for all image-size & channels. Image parameter are obtained from the X_train variable loaded by utils_mnist.py

Conformed to PEP8

### CODE REVIEW

**Comment by aashish-kumar** [comment]
The auto-build has not started for sometime. please check.

**Comment by aashish-kumar** [comment]
Changelog:-
1) Removed Hard-coded values and used the info from dataset to populate them:
      - Images:  (28,28,1)  to (img_rows,img_cols,nchannels).
      - Number of classes (10) to nb_classes  (some tutorials had it hard coded it while others were not using it for model generation)
 2)  Commented `assert X_test.shape[0] == test_end - test_start, X_test.shape`  : The assertion is too specific to MNIST 
3)  Made Label Smoothing optional(added an argument) as some tutorials are using it while others do not.
4) In Keras Tutorial: Changed the model dump from /tmp to local folder. This can be problem when multiple users on the same machine :- This one is debatable.
5) Removed the flag `-    flags.DEFINE_integer('nb_classes', 10, 'Number of output classes')` as changing this parameter doesn't actually change the model parameters. 

Comments?  @npapernot

File: cleverhans_tutorials/mnist_tutorial_cw.py [hunk 1]
```diff
@@ -104,7 +107,7 @@ def mnist_tutorial_cw(train_start=0, train_end=60000, test_start=0,
     # Evaluate the accuracy of the MNIST model on legitimate test examples
     eval_params = {'batch_size': batch_size}
     accuracy = model_eval(sess, x, y, preds, X_test, Y_test, args=eval_params)
-    assert X_test.shape[0] == test_end - test_start, X_test.shape
+#    assert X_test.shape[0] == test_end - test_start, X_test.shape
```
Comment (goodfeli) [inline cleverhans_tutorials/mnist_tutorial_cw.py]: What's going on here?

If there's a good reason to remove the test, it's fine to remove.

If there's no reason to remove it, I'd prefer to keep it. A bit of defensive programming helps to prevent bugs and catch them quickly when they arise.

We generally don't want commented code in the repo.

File: cleverhans_tutorials/mnist_tutorial_jsma.py [hunk 2]
```diff
@@ -67,12 +62,20 @@ def mnist_tutorial_jsma(train_start=0, train_end=60000, test_start=0,
                                                   test_start=test_start,
                                                   test_end=test_end)
 
+    print(X_train.shape)
```
Comment (goodfeli) [inline cleverhans_tutorials/mnist_tutorial_jsma.py]: Let's not add debugging prints to the repo

File: cleverhans_tutorials/mnist_tutorial_jsma.py [hunk 3]
```diff
@@ -94,7 +97,7 @@ def mnist_tutorial_jsma(train_start=0, train_end=60000, test_start=0,
     # Evaluate the accuracy of the MNIST model on legitimate test examples
     eval_params = {'batch_size': batch_size}
     accuracy = model_eval(sess, x, y, preds, X_test, Y_test, args=eval_params)
-    assert X_test.shape[0] == test_end - test_start, X_test.shape
+#    assert X_test.shape[0] == test_end - test_start, X_test.shape
```
Comment (goodfeli) [inline cleverhans_tutorials/mnist_tutorial_jsma.py]: Same issue with commented test here

File: cleverhans_tutorials/mnist_tutorial_tf.py [hunk 4]
```diff
@@ -75,15 +76,20 @@ def mnist_tutorial(train_start=0, train_end=60000, test_start=0,
                                                   train_end=train_end,
                                                   test_start=test_start,
                                                   test_end=test_end)
-
     # Use label smoothing
-    assert Y_train.shape[1] == 10
-    label_smooth = .1
-    Y_train = Y_train.clip(label_smooth / 9., 1. - label_smooth)
+    print(X_train.shape)
```
Comment (goodfeli) [inline cleverhans_tutorials/mnist_tutorial_tf.py]: Debugging print

**Comment by goodfeli** [comment]
LGTM once the commented code / print statements are fixed

File: cleverhans_tutorials/mnist_tutorial_cw.py (hunk 1 above)
Comment (aashish-kumar) [reply cleverhans_tutorials/mnist_tutorial_cw.py]: I commented this line as test_end and test_start are hard coded and depend on the dataset(number of training/test samples) hence making the code specific to MNIST. I am not sure for the purpose of having test_start and test_end. I felt it might be to control number of samples getting used but then its index specific too.

Comment (goodfeli) [reply cleverhans_tutorials/mnist_tutorial_cw.py]: The user can pass in any number for `test_start` and `test_end` so they're not specific to MNIST. The idea is that you can evaluate on just a subset of the test set if you want your evaluation to run fast. I think we use this in the tests.

Comment (aashish-kumar) [reply cleverhans_tutorials/mnist_tutorial_cw.py]: I am keeping the assert as of now. Can you please elaborate on the kind of bugs this assert is planned to detect? Will those be data loading bugs or model evaluation bugs?

I feel we should also keep an assert for test_end < len(test_data) immediately after the data loading.

**Comment by aashish-kumar** [comment]
Resolved conflict with latest commit.
There is an interesting update.
X_train,Y_train have been changed to x_train,y_train. Is this change planned?
I have seen that these variables names are not consistent for all the tests. I have fixed most of them as of now.

**Comment by npapernot** [comment]
Yes that was a recent change, variable names were changed from X_train to x_train in #423

**Comment by goodfeli** [comment]
Thanks!
